### DISCUSSION TITLE
me/50 cant change default branch cant delete branch marked as default so i cant finish my final project.

### DISCUSSION THREAD

**Discussion by AvatharBehemoth** [community/community#66042] [Repositories] [answered]
### Select Topic Area

Question

### Body

Im trying to finish my course CS50G. So i need to upload my project to https://github.com/me50/USERNAME.git, which if you visit does NOT show my project. Only by visiting the branch marked as "Master" do i see my project. The page: https://github.com/me50/USERNAME/blob/games50/projects/2018/x/final/ doesn not mirror the project. 
It has the label "default" and sometimes "Default" which i cant set to a different branch. I deleted https://github.com/me50/USERNAME/blob/games50/projects/2018/x/final/ but it did not dissapear from the list. gitbash cant find it though. 
So it simultaniously does and doesnt exist. In the "view all branches page" it misses ALL the buttons except "view branch activity". 

So i hope can help me. This seems like something broke in github. Anyway i need to put my project there. Please help.

**Comment by AvatharBehemoth** [comment]
I just uploaded the files manually, using the gitignore template for Unity. But it sucks.
I got some weird 404 statusses after hitting the commit button after uploading. So my directory truly both existed and didnt exist at the same time. 
Also for some reason the gitignore file stipulated to ignore directories such as "Usersettings", "obj", "Logs" etc but gitbash uploaded them anyway??!!?? Seems like that gitignore really works.

**Comment by seyLu** [comment, marked as answer]
#### Hi @AvatharBehemoth 

> Im trying to finish my course CS50G. So i need to upload my project to `https://github.com/me50/USERNAME.git`, which if you visit does NOT show my project. Only by visiting the branch marked as "Master" do i see my project. The page: `https://github.com/me50/USERNAME/blob/games50/projects/2018/x/final/` 

- Did you use submit50? If not, then git by default, if you are in the master/main branch in local, will push changes to master/main branch in remote -- unless you explicitly create a new branch, make changes, and push the changes to a remote branch of the same name. [[Manually submit CS50g proj]](https://www.reddit.com/r/cs50/comments/k5ksp8/how_to_submit_mario_and_pong_in_cs50_games_track/)

- If you used submit50, IIRC this is a known bug if you accidentally mess with the default branch. [[Problem with submitting to CS50AI]](https://www.reddit.com/r/cs50/comments/jqpeq9/a_problem_when_trying_to_submit_my_first_cs50ai/) 

<br>

>  I deleted `https://github.com/me50/USERNAME/blob/games50/projects/2018/x/final/` but it did not dissapear from the list. 

- It takes a while for the branch UI of github to update (also make sure cache is clear, or do a hard refresh)

<br>

> Also for some reason the gitignore file stipulated to ignore directories such as "Usersettings", "obj", "Logs" etc but gitbash uploaded them anyway??!!?? Seems like that gitignore really works.

- the .gitignore template for unity assumes the file structure as:

```bash
Assets
Library
Logs
...
.gitignore
```

- and not

```bash
# in your case:
ProjectName
 |-- Assets
 |-- Library
 |-- Logs
...
.gitignore
```

- which would cause you to upload large files (which github is not really meant for and could easily be solved with the proper directory structure)

- also make sure when issuing git commands, you are in the root directory.

**Comment by AvatharBehemoth** [comment]
Hey seylu, thanks for your answer! 
I succeeded in submitting (as you can read in my own reply). but it was the ugly solution. The thing is, i deleted the "default" directory/url. but what happened was that the url did not dissapear but the icons next to it did (in "view all branches"). So like i said the url simultaniously didnt and did exist (schroedingers url). I think this is a system level design error. Which normally you dont have (because you have all the rights) but when using me50, you dont have all the rights! So you can get stuck in limbo!
